# Prefered column or line import (`turmag-special-rules/import-entities-by-column-or-line`)

ðŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->

## ðŸ“– Rule Details

This rule regulates imports in a column or in a line, depending on the passed parameter **minProperties**, which is responsible for how many parameters are used to configure imports in a column

```vue
<script>
// For example, the passed property minProperties is 3 - line to column
// âœ— BAD
import { ref, computed, watch, onMounted } from 'vue';
import { UiButton, UiSwitch, UiToggler } from 'somepath/kit';
import { declOfNum, numberFormatter, getPath } from 'somepath/helpers';

// âœ“ GOOD
import {
    ref,
    computed,
    watch,
    onMounted,
} from 'vue';
import {
    UiButton,
    UiSwitch,
    UiToggler,
} from 'somepath/kit';
import {
    declOfNum,
    numberFormatter,
    getPath,
} from 'somepath/helpers';
</script>
```

```vue
<script>
// For example, the passed property minProperties is 3  - column to line
// âœ— BAD
import {
    ref,
    computed,
} from 'vue';
import {
    UiButton,
    UiSwitch,
} from 'somepath/kit';
import {
    declOfNum,
    numberFormatter,
} from 'somepath/helpers';

// âœ“ GOOD
import { ref, computed } from 'vue';
import { UiButton, UiSwitch } from 'somepath/kit';
import { declOfNum, numberFormatter } from 'somepath/helpers';
</script>
```

## ðŸ”§ Rule Details (Config)
You can activate it rule or not.
But there is important setting - aliases.


```js
{
  "import-entities-by-column-or-line": ["error", { minProperties: 3 }]
}
```

## ðŸ”Ž Implementation

- [Rule source](../../lib/rules/import-entities-by-column-or-line.js)